packages/flutter_tools/lib/src/base/terminal.dart

```diff
@@ -187,6 +191,16 @@ class AnsiTerminal implements Terminal {
   bool get supportsEmoji => !_platform.isWindows
     || _platform.environment.containsKey('WT_SESSION');
 
+  @override
+  int get preferredStyle {
+    final DateTime now = DateTime.now();
```

Comment:
I love this but the DateTime.now is going to make it difficult to test for real. As a testing workaround, you could give the constructor a test-only named parameter for overriding:

```dart
  AnsiTerminal({
    required io.Stdio stdio,
    required Platform platform,
    @visibleForTesting CurrentTime currentTime = DateTime.now // Maybe this is a constructor and can't be torn off yet, but something like this
  })
```

Reply:
Good call.

Reply:
Ok I went with something like this (except I just take a DateTime, not a Clock) and used it to test preferredStyle, which I hadn't figured out how to test before. Also this allowed me to actually make the tests pass since it turns out I'd totally flaked on thinking through how that would work before heh.

Reply:
Oh but having it required breaks google3, I guess I'll have to make it default to something.